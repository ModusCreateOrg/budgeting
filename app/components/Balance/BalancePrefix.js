// @flow

import * as React from 'react';

import styles from './style.scss';

type BalancePrefixProps = {
  text: string,
};

const BalancePrefix = ({ text }: BalancePrefixProps) =>
  <div className={styles.balanceSymbol}>
    {text}
  </div>;

BalancePrefix.defaultProps = {
  text: '',
};

export default BalancePrefix;
