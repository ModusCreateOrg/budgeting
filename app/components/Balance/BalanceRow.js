// @flow
import * as React from 'react';

import styles from './style.scss';

type BalanceProps = {
  children: React.Node,
};

const BalanceRow = ({ children }: BalanceProps) => (
  <div className={styles.balanceRowContainer}>
    <div className={styles.balanceRow}>{children}</div>
  </div>
);

export default BalanceRow;
