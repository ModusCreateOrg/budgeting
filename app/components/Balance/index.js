// @flow

import * as React from 'react';
import type { FormattedAmount } from 'utils/formatAmount';

import styles from './style.scss';

type BalanceProps = {
  title: string,
  amount: FormattedAmount,
  colorize: boolean,
  prefix: ?string,
};

const Balance = ({ title, amount, colorize, prefix }: BalanceProps) => {
  const amountCls = colorize && amount.isNegative ? styles.neg : styles.pos || '';

  const prefixElement = typeof prefix === 'string' && (
    <div key="prefix" className={styles.balanceSymbol}>
      {prefix}
    </div>
  );

  const balanceElement = (
    <div key="item" className={styles.balanceWrapper}>
      <div className={styles.balanceItem}>
        <div className={`${styles.balanceAmount} ${amountCls}`}>{amount.text}</div>
        <div className={styles.balanceTitle}>{title}</div>
      </div>
    </div>
  );
  return [prefixElement, balanceElement];
};

Balance.defaultProps = {
  colorize: true,
  prefix: null,
};

export default Balance;
