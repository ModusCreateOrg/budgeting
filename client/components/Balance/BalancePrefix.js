import React, { PropTypes } from 'react';

import './style.scss';

const BalancePrefix = ({ text }) => <div className="balance-symbol">{text}</div>;

BalancePrefix.propTypes = {
  text: PropTypes.string.isRequired
};

export default BalancePrefix;
