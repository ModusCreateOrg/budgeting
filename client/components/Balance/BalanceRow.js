import React, { PropTypes } from 'react';

import styles from './style.scss';

const Balance = ({ children }) => (
  <div className={styles.balanceRowContainer}>
    <div className={styles.balanceRow}>
      {children}
    </div>
  </div>
);

Balance.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Balance;
