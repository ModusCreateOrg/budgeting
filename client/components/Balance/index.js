import React, { PropTypes } from 'react';

import styles from './style.scss';

const Balance = ({ title, amount, colorize }) => {
  const amountCls = colorize && (amount.isNegative ? styles.neg : styles.pos);

  return (
    <div className={styles.balanceWrapper}>
      <div className={styles.balanceItem}>
        <div className={`${styles.balanceAmount} ${amountCls}`}>{amount.text}</div>
        <div className={styles.balanceTitle}>{title}</div>
      </div>
    </div>
  );
};

Balance.propTypes = {
  title: PropTypes.string.isRequired,
  amount: PropTypes.object.isRequired,
  colorize: PropTypes.bool
};

Balance.defaultProps = {
  colorize: true
};

export default Balance;
