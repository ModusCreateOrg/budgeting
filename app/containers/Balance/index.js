// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import type { FormattedAmount } from 'utils/formatAmount';
import { getFormattedBalance, getFormattedInflowBalance, getFormattedOutflowBalance } from 'selectors/transactions';

import BalanceItem from 'components/Balance';
import BalanceRow from 'components/Balance/BalanceRow';

type BalanceProps = {
  balance: FormattedAmount,
  inflow: FormattedAmount,
  outflow: FormattedAmount,
};

class Balance extends React.Component<BalanceProps> {
  render() {
    const { inflow, outflow, balance } = this.props;

    return (
      <BalanceRow>
        <BalanceItem amount={inflow} title="Total Inflow" />
        <BalanceItem amount={outflow} title="Total Outflow" prefix="-" />
        <BalanceItem amount={balance} title="Working Balance" colorize={false} prefix="=" />
      </BalanceRow>
    );
  }
}

const mapStateToProps = state => ({
  balance: getFormattedBalance(state),
  inflow: getFormattedInflowBalance(state),
  outflow: getFormattedOutflowBalance(state),
});

export default connect(mapStateToProps)(Balance);
