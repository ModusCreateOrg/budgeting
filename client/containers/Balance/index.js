import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import {
  getFormattedBalance,
  getFormattedInflowBalance,
  getFormattedOutflowBalance
} from 'modules/transactions';

import BalanceItem from 'components/Balance';
import BalancePrefix from 'components/Balance/BalancePrefix';
import BalanceRow from 'components/Balance/BalanceRow';

@connect(
  state => ({
    balance: getFormattedBalance(state),
    inflow: getFormattedInflowBalance(state),
    outflow: getFormattedOutflowBalance(state)    
  })
)
class Balance extends Component {
  static propTypes = {
    balance: PropTypes.object.isRequired,
    inflow: PropTypes.object.isRequired,
    outflow: PropTypes.object.isRequired
  }

  render() {
    const { inflow, outflow, balance } = this.props;

    return (
      <BalanceRow>
        <BalanceItem amount={inflow} title="Total Inflow" />
        <BalancePrefix text="-" />
        <BalanceItem amount={outflow} title="Total Outflow" />
        <BalancePrefix text="=" />
        <BalanceItem amount={balance} title="Working Balance" colorize={false} />
      </BalanceRow>
    );
  }
}

export default Balance;
