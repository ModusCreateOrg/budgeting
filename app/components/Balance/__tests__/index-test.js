import React from 'react';
import renderer from 'react-test-renderer';
import Balance from '../';

it('renders correctly', () => {
  const mockAmount = {
    text: 'text',
    isNegative: false,
  };
  const tree = renderer.create(<Balance title="title" amount={mockAmount} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with a prefix', () => {
  const mockAmount = {
    text: 'text',
    isNegative: false,
  };
  const tree = renderer.create(<Balance title="title" amount={mockAmount} prefix="prefix" />).toJSON();
  expect(tree).toMatchSnapshot();
});
