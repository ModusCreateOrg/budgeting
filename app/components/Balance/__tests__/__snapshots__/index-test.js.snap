// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="balanceWrapper"
>
  <div
    className="balanceItem"
  >
    <div
      className="balanceAmount pos"
    >
      text
    </div>
    <div
      className="balanceTitle"
    >
      title
    </div>
  </div>
</div>
`;

exports[`renders correctly with a prefix 1`] = `
Array [
  <div
    className="balanceSymbol"
  >
    prefix
  </div>,
  <div
    className="balanceWrapper"
  >
    <div
      className="balanceItem"
    >
      <div
        className="balanceAmount pos"
      >
        text
      </div>
      <div
        className="balanceTitle"
      >
        title
      </div>
    </div>
  </div>,
]
`;
